packages/@aws-cdk/aws-eks/lib/gs-extension/cluster-resource-nested-stack.ts

```diff
@@ -0,0 +1,62 @@
+import * as ec2 from '@aws-cdk/aws-ec2';
+import * as iam from '@aws-cdk/aws-iam';
+import { CfnStack, Fn, RemovalPolicy, Token } from '@aws-cdk/core';
+
+// v2 - keep this import as a separate section to reduce merge conflict when forward merging with the v2 branch.
+// eslint-disable-next-line
+import { Construct as CoreConstruct } from '@aws-cdk/core';
+
+export interface ClusterResourceNestedStackProps {
+  templateURL: string;
+  clusterCreationRole: iam.IRole;
+  subnets?: ec2.ISubnet[];
+  securityGroup?: ec2.ISecurityGroup;
+
+  removalPolicy?: RemovalPolicy;
+}
+
+const PROVIDER_ARN_OUTPUT_NAME = 'Outputs.ClusterResourceProviderframeworkonEventC6B02E13Arn';
```

Comment:
Just leaving a comment to ensure the resource output of Lambda matches the ARN of the stack